### PR TITLE
CI build template

### PR DESCRIPTION
**Pull request by aminya**
The build template cherry-picked from https://github.com/atom/atom/pull/21081.

See that pull request for more information.

### CODE REVIEW

File: script/vsts/platforms/templates/build.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      } else {
+        script/build --no-bootstrap --compress-artifacts $esc $env:LinuxArgs $CodeSign $CreateWindowsInstallaer
+      }
+    displayName: Build Atom
+    env:
+      GITHUB_TOKEN: $(GITHUB_TOKEN)
+      IS_RELEASE_BRANCH: $(IsReleaseBranch)
+      IS_SIGNED_ZIP_BRANCH: $(IsSignedZipBranch)
+      ATOM_RELEASE_VERSION: $(ReleaseVersion)
+
```
Comment (DeeDeeG) [inline script/vsts/platforms/templates/build.yml:53]: In this `env:` block, we should additionally be passing in all of the following:
```yml
          ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL: $(ATOM_MAC_CODE_SIGNING_CERT_DOWNLOAD_URL)
          ATOM_MAC_CODE_SIGNING_CERT_PASSWORD: $(ATOM_MAC_CODE_SIGNING_CERT_PASSWORD)
          ATOM_MAC_CODE_SIGNING_KEYCHAIN: $(ATOM_MAC_CODE_SIGNING_KEYCHAIN)
          ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD: $(ATOM_MAC_CODE_SIGNING_KEYCHAIN_PASSWORD)
          ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL: $(ATOM_WIN_CODE_SIGNING_CERT_DOWNLOAD_URL)
          ATOM_WIN_CODE_SIGNING_CERT_PASSWORD: $(ATOM_WIN_CODE_SIGNING_CERT_PASSWORD)

**Review by DeeDeeG — COMMENTED** [review]
Sorry, we missed something here. It wasn't obvious to us, since we never code sign over at our fork.

cc @sadick254 this suggested change should get builds passing again on the Production and Nightly pipelines. (Also: without this change, builds are not being code-signed on macOS and Windows.)

I can open a PR to address this.

**Comment by sadick254** [comment]
@DeeDeeG sure, that would be great

**Comment by aminya** [comment]
I made a patch in https://github.com/atom/atom/pull/21264.

**Comment by DeeDeeG** [comment]
@sadick254 The PR @aminya posted looks good to me. 👍

**Pull request merged**